### DISCUSSION TITLE
GSOC 2021 Community Detection Algorithms

### DISCUSSION THREAD

**Discussion by Clearloveyuan** [networkx/networkx#4703] [Q&A]
Hello, Everybody

My name Dongyuan Li and I am a Ph.D. student at the Tokyo Institute of Technology. I am a CS student and my research interests are community detection and big data analysis.

I am good at python and Matlab, and I propose some dynamic community detection methods and share codes on my Github.

I have read the paper about Louvain Community Detection:

Blondel, Vincent D., et al. "Fast unfolding of communities in large networks." Journal of statistical mechanics: theory and experiment 2008.10 (2008): P10008.

And I have used the Louvain Community Detection method on my project. The following is a version of the Louvain Community Detection method on my GitHub:
https://github.com/a626709452/Louvain-Community-detection/blob/main/Louvain.py

I would be glad to get a mentor with whom we can discuss the possibilities of participating in these projects.

**Comment by MridulS** [comment]
Hi!

There have been multiple attempts on Louvain before so it would be nice if you can go through https://github.com/networkx/networkx/pull/1090, https://github.com/networkx/networkx/pull/1092 and https://github.com/networkx/networkx/issues/951 to get a better picture of integrating Louvain into NetworkX.

Also it would be nice to get a comparison between your implementation https://github.com/a626709452/Louvain-Community-detection/blob/main/Louvain.py and https://github.com/taynaud/python-louvain (people usually use this when they want Louvain with NetworkX).
